### PR TITLE
Login

### LINKED ISSUE #8: Create Registration Page
Ask user for 
username,
email
password
name

### LINKED ISSUE #9: Create Login Page
Redirect to home page after successful login

### PR DESCRIPTION
**Pull request by rbhastie**
## What?
#9 Login form component

## Why?
For users to log into the app and gain access to the user dashboard

## How?
React Login component with MUI integration and useState hook

## Testing?
No tests have been implemented in this commit

## Anything Else?
This form is based on the MUI template found [here](https://github.com/mui/material-ui/tree/v5.5.2/docs/data/material/getting-started/templates/sign-in)

### CODE REVIEW

**Comment by rbhastie** [comment]
Edited to merge to dev branch. Mistakenly originally created for main branch, please reject if there are any concerns.

File: client/App.jsx [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  Link,
+} from 'react-router-dom';
 
 import './styles/styles.css';
 
-const App = props => {
+class App extends Component {
```
Comment (shalarewicz) [inline client/App.jsx]: This is stateless so could be a functional component

File: client/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import Container from '@mui/material/Container';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+
+
+
+const theme = createTheme();
+
+export default function Login() {
+  const [inputs, setInputs] = useState({
+    email: '',
```
Comment (shalarewicz) [inline client/Login.jsx:20]: Use username here or allow login with email or username

File: client/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  const handleChange = (event) => {
+    const name = event.target.name;
+    const value = event.target.value;
+    setInputs(values => ({...values, [name]: value}));
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const data = new FormData(document.getElementById('login'));
+    console.log({
```
Comment (shalarewicz) [inline client/Login.jsx:33]: Can remove this

File: client/SignUp.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+const theme = createTheme();
+
+export default function SignUp() {
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    console.log({
+      email: data.get('email'),
+      password: data.get('password'),
+    });
```
Comment (shalarewicz) [inline client/SignUp.jsx:52]: Add fetch request here

**Review by shalarewicz — CHANGES_REQUESTED** [review]
Add fetch request for registration form submission. Maybe POST /api/register?
Other minor stylistic comments

File: client/App.jsx [hunk 5]
```diff
@@ -1,17 +1,47 @@
...
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+
+    return (
```
Comment (shalarewicz) [inline client/App.jsx:20]: Let's use the <BrowserRouter> component here. It's better for keeping the UI in sync with the URL

File: client/Login.jsx (hunk 2 above)
Comment (rbhastie) [reply client/Login.jsx:20]: We agreed to eliminate username for login and from the user table of the database.

Comment (shalarewicz) [reply client/Login.jsx:20]: ok sounds good

**Review by shalarewicz — APPROVED** [review]
Great work!

**Pull request merged**